core/iwasm/libraries/lib-pthread/lib_pthread_wrapper.c

```diff
@@ -527,9 +527,6 @@ pthread_start_routine(void *arg)
     /* destroy pthread key values */
     call_key_destructor(exec_env);
 
-    /* routine exit, destroy instance */
-    wasm_runtime_deinstantiate_internal(module_inst, true);
-
```

Comment:
how is this related to the rest of the PR?

Reply:
If `WASM_ENABLE_HEAP_AUX_STACK_ALLOCATION` is enabled, module instance can't be deinstantiate untill the memory allocated for the aux stack is released.

Reply:
* with this change, where will it be deinstantiated instead?
* how about the other wasm_runtime_deinstantiate_internal call in this function? (L507)

Reply:
> with this change, where will it be deinstantiated instead?
It's deinstantiated in the `thread_manager_start_routine()`

> how about the other wasm_runtime_deinstantiate_internal call in this function? (L507)
Yes, this one, and one in the `lib_wasi_threads_wrapper.c` have to be removed. Already updated the code, thanks.

Reply:
do you mean already updated locally but not pushed yet?

Reply:
Yes sorry, didn't push my change. Should be updated now.